### PR TITLE
Yaling - L145.water.demand.municipal

### PR DESCRIPTION
**Pull request by yalingliupnnl**
The format mismatch between the old and new systems for the file L145.municipal_water_R_W_Yh_km3.csv has not been addressed yet, need help.

#312

### CODE REVIEW

File: R/zchunk_L145.water.demand.municipal.R [hunk 1]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
+    # Aggregate FAO_municipal_water_AQUASTAT to GCAM regions and fill out the years for missing data.
+
+    #Map FAO country to GCAM region ID
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Throughout: space after `#`

File: R/zchunk_L145.water.demand.municipal.R [hunk 2]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+    iso_GCAM_regID %>% select(iso, GCAM_region_ID) %>%
+      inner_join(AGLU_ctry[,c("iso","FAO_country")], by ="iso") -> FAO_ctry_GCAM_region_ID
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Throughout: surround mathematical operators like `=` with spaces

File: R/zchunk_L145.water.demand.municipal.R [hunk 3]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+
+    FAO_municipal_water_AQUASTAT %>%
+      rename ( FAO_country =Area ) %>%
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: No GCAM-style spacing with parentheses please

File: R/zchunk_L145.water.demand.municipal.R (hunk 2 above)
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Perhaps `select(AGLU_ctry, iso, FAO_country)` to be consistent

File: R/zchunk_L145.water.demand.municipal.R [hunk 4]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+      ungroup() %>%
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+               Year = HISTORICAL_YEARS,
+               fill = list(Value = NA))%>%
+      group_by(GCAM_region_ID)%>%
+      mutate(Value = approx_fun(Year, Value, rule=2)) %>%
+      rename(value = Value, year =Year) %>%
+      mutate(water_type ="water withdrawals")-> municipal_water_R_W_Yh_km3
+
+    # Come up with GCAM reginal average prices starting with the country level IBNET data.
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: "regional"

File: R/zchunk_L145.water.demand.municipal.R [hunk 5]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+    # Note since the years are all over the place we will just use the average across years too.
+    #Come up with regional municipal prices from IBNET data
+    IBNET_municipal_water_cost_USDm3 %>%
+      mutate(FAO_country = country)%>%
+      inner_join(FAO_ctry_GCAM_region_ID, by="FAO_country")%>%
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Two lines above could be (I think) simply `inner_join(FAO_ctry_GCAM_region_ID, by = c("country" = "FAO_country"))`

File: R/zchunk_L145.water.demand.municipal.R [hunk 6]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+
+    # The IBNET data is incomplete and so it is very possible that we have entire GCAM regions in which none
+    # of the contained countries had any reported data.
+    stopifnot( nrow( municipal_water_cost_R_75USD_m3 ) == nrow( GCAM_region_names ) )
+
+    # Map water use efficiencies from the continent scale to GCAM regions
+    # Note names were changed to match manufacturing continent to avoid the need for an additional mapping
+
+    #create the variable year2 for temporal interpolation
+    year2 <- sort(c(1900, 1940, 1950, 1960, 1970, 1980, 1990, 1995, 2000,modeltime.YEARS[-2]))
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Very confusing what's going on here. What are these years? Why are you dropping the second element of `modeltime.YEARS`?

File: R/zchunk_L145.water.demand.municipal.R [hunk 7]
```diff
@@ -66,36 +124,48 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
     # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
     # There's also a `same_precursors_as(x)` you can use
     # If no precursors (very rare) don't call `add_precursor` at all
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Remove all these comment lines

File: R/zchunk_L145.water.demand.municipal.R [hunk 8]
```diff
@@ -66,36 +124,48 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
+    municipal_water_R_W_Yh_km3 %>%
+      add_title("Municipal water withdrawals by GCAM_region_ID for all historical years ") %>%
+      add_units("km^3") %>%
+      add_comments("Aggregate FAO_municipal_water_AQUASTAT to GCAM regions and fill out the years for missing data by rule=2") %>%
       add_comments("can be multiple lines") %>%
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Remove

File: test.log [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Wed May 24 10:37:56 2017  Opening test.log
```
Comment (bpbond) [inline test.log]: What is this file? Remove from commit please

File: R/zchunk_L145.water.demand.municipal.R [hunk 10]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+
+    #create the variable year2 for temporal interpolation
+    year2 <- sort(c(1900, 1940, 1950, 1960, 1970, 1980, 1990, 1995, 2000,modeltime.YEARS[-2]))
+
+    #Map municipal water use efficiencies to GCAM regions
+    municipal_water_use_efficiency %>%
+      gather(variable,value,-continent)%>%
+      mutate(variable = as.numeric(variable))%>%
+      rename (year=variable) %>%
+      inner_join(manufacturing_water_mapping, by = "continent") %>%
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: Why `inner_join`? Can you use `left_join_error_no_match` instead? Same comment applies to other inner joins.

File: R/zchunk_L145.water.demand.municipal.R [hunk 11]
```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
...
+      inner_join(GCAM_region_names, by = "region") %>%
+      select(-continent, -GCAM_region_ID) %>%
+      complete(region = unique(GCAM_region_names$region),
+               year = year2,
+               fill = list(value = NA)) %>%
+      group_by(region) %>%
+      #Fillout the coefficients for all years using rule=2
+      mutate(coefficient = approx_fun(year, value, rule=2)) %>%
+      left_join_keep_first_only(GCAM_region_names, by="region")%>%
+      filter(year>=1975, year != 1980, year !=1995, year !=2000)%>%
```
Comment (bpbond) [inline R/zchunk_L145.water.demand.municipal.R]: What are these hardcoded year values? We really don't want to do this.

**Comment by bpbond** [comment]
@yalingliupnnl It looks like there are actually some significant differences between the datasets:

```R
> newdata  <- x[["L145.municipal_water_R_W_Yh_km3"]] 
> olddata <- readr::read_csv("tests/testthat/comparison_data/water/L145.municipal_water_R_W_Yh_km3.csv", comment="#")
> olddata <- tidyr::gather(olddata, year, old.value, -1:-2)
> olddata$year <- as.integer(substr(olddata$year, 2, 5))
> comb <- merge(olddata, newdata)
> comb$diff <- comb$value-comb$old.value
> summary(comb)
 GCAM_region_ID   water_type             year        old.value          value             diff         
 Min.   : 1.00   Length:1280        Min.   :1971   Min.   : 0.001   Min.   : 0.001   Min.   :-1.63294  
 1st Qu.: 8.75   Class :character   1st Qu.:1981   1st Qu.: 0.930   1st Qu.: 0.874   1st Qu.: 0.00000  
 Median :16.50   Mode  :character   Median :1990   Median : 2.970   Median : 2.712   Median : 0.00000  
 Mean   :16.50                      Mean   :1990   Mean   : 7.863   Mean   : 7.955   Mean   :-0.00697  
 3rd Qu.:24.25                      3rd Qu.:2000   3rd Qu.: 9.600   3rd Qu.: 9.600   3rd Qu.: 0.00000  
 Max.   :32.00                      Max.   :2010   Max.   :76.600   Max.   :76.600   Max.   : 1.05280  
                                                                    NA's   :40       NA's   :40        
```

**Comment by bpbond** [comment]
...more specifically, it looks like the new dataset has 40 `NA` values that aren't present in the old one.

**Comment by yalingliupnnl** [comment]
@bpbond Thanks Ben, let me check it first.

**Comment by bpbond** [comment]
As this PR doesn't seem to be a priority, I'm closing it. Feel free to re-open when you have time to deal with it. cc @mihejazi

**Pull request closed without merging**